compiler/phpdoc.h

```diff
@@ -4,81 +4,116 @@
 
 #pragma once
 
-#include <map>
-#include <optional>
 #include <string>
-#include <vector>
+#include <forward_list>
 
 #include "compiler/data/data_ptr.h"
 #include "compiler/data/vertex-adaptor.h"
-#include "compiler/inferring/primitive-type.h"
-
-struct php_doc_tag {
-  enum doc_type {
-    unknown,
-    kphp_inline,
-    kphp_infer,
-    kphp_disable_warnings,
-    kphp_extern_func_info,
-    kphp_pure_function,
-    param,
-    returns,
-    var,
-    kphp_required,
-    kphp_lib_export,
-    kphp_sync,
-    kphp_should_not_throw,
-    kphp_throws,
-    kphp_template,
-    kphp_param,
-    kphp_return,
-    kphp_memcache_class,
-    kphp_immutable_class,
-    kphp_tl_class,
-    kphp_const,
-    kphp_noreturn,
-    kphp_warn_unused_result,
-    kphp_warn_performance,
-    kphp_analyze_performance,
-    kphp_flatten,
-    kphp_serializable,
-    kphp_reserved_fields,
-    kphp_serialized_field,
-    kphp_serialized_float32,
-    kphp_profile,
-    kphp_profile_allow_inline,
-    kphp_strict_types_enable, // TODO: remove when strict_types=1 are enabled by default
-    kphp_color,
+
+
+enum class PhpDocType {
+  unknown,
+  kphp_inline,
+  kphp_infer,
+  kphp_disable_warnings,
+  kphp_extern_func_info,
+  kphp_pure_function,
+  param,
+  returns,
+  var,
+  kphp_required,
+  kphp_lib_export,
+  kphp_sync,
+  kphp_should_not_throw,
+  kphp_throws,
+  kphp_template,
+  kphp_param,
+  kphp_return,
+  kphp_memcache_class,
+  kphp_immutable_class,
+  kphp_tl_class,
+  kphp_const,
+  kphp_noreturn,
+  kphp_warn_unused_result,
+  kphp_warn_performance,
+  kphp_analyze_performance,
+  kphp_flatten,
+  kphp_serializable,
+  kphp_reserved_fields,
+  kphp_serialized_field,
+  kphp_serialized_float32,
+  kphp_profile,
+  kphp_profile_allow_inline,
+  kphp_strict_types_enable, // TODO: remove when strict_types=1 are enabled by default
+  kphp_color,
+};
+
+// one phpdoc contains of several doc tags
+// every doc tag has a type (enum mapping @tag-name) and a value
+struct PhpDocTag {
+  // if a tag value is like "int|false $var_name" (for @param / @var / etc.),
+  // it's represented as this structure
+  struct TypeAndVarName {
+    const TypeHint *type_hint{nullptr};
+    vk::string_view var_name;     // stored without leading "$"; could be empty if omitted in phpdoc
```

Comment:
std::string_view here?

Reply:
Maybe. Could you please advise, how to replace `value.starts_with("something")` if value is std::string_view?

Reply:
Ah, in this case nothing. It will be possible after debian buster, or if we upgrade compiler there.